README.md

```diff
@@ -470,40 +470,36 @@ membership".
 
 ## 4.10. Tangles
 
-A "tangle" in scuttlebutt is a way to define a directed acyclic graph (DAG) of 
+A "tangle" in scuttlebutt is a way to define a directed acyclic graph (DAG) of
 messages. You can read more about them in the [Tangle SIP].
 
 
 ### 4.10.1. Members tangle
 
-The purpose of this tangle is to track the group membership (for a  particular 
-epoch of the group).
-
-We define the members tangle for some message ID `A`as:
-1. **candidate messages**:
-   - the root message with id `A` so long as it has
-     ```javascript
-     content.tangles.members = { root: null, previous: null }
-     ```
-   - any messages where
-     ```javascript
-     content.tangles.members = { root: A, previous: [...] }
-     ```
-2. **recipe**:
-    - a) the graph starts with the root message
-    - b) for the current tip(s) of the tangle
-        - find candidate messages which list that tip in the `previous` field   
-          of their tangle data
-        - check that the graph contains each message in `previous`
-            - if it does, add it to the graph
-            - otherwise set it aside till the graph does contain them
-        - newly added messages to the tangle graph constitue new tips
-    - c) repeat (b) till you cannot grow the tangle further
-
-
-In the context of private groups, the addition to a particular epoch is always
-published to an earlier epoch (or in the case of the first epoch, to the
-Additions feed).
+The purpose of the members tangle is to group and partially order all messages
+that alter the group membership during a single epoch.  A peer can calculate
+the Set of members in an epoch by [reducing] the messages in this tangle.  Each
+epoch will have its own members tangle.  The human-friendly name in the tangle
+data is the string `members`.
+
+We construct the members tangle for some epoch `E` of a group `G` as:
+
+1. **Candidate messages**:
+    - The message with `content.type` equal to `group/init` which started epoch
+    `E`
+    - Any message published on the epoch feed for `E` where `content.type` is
+    equal to `group/add-member`
+    - Any message published on the epoch feed for `E` where `content.type` is
+    equal to `group/exclude`
+2. **Recipe**:
+    - If the candidate message has `content.type` equal to `group/init`, then
+    the tangle data MUST be `{"root": null, "previous": null}`.
+    - Otherwise, the candidate message's tangle data MUST have `root` equal to
+    the ID of the root message `R`, and `previous` SHOULD be an array of message
+    IDs for all locally known "tips" of the tangle
```

Comment:
To me, these are candidacy requirements. The "recipe" is how you take candidates and build a tangle from them.
- do you just follow the "basic" recipe?
- are there any additional checks you run while extending the graph

The candidacy for me is about more than the type (in the code). I drop any malformed messages - e.g. if you don't have a recps field on `group/add-member` then that will not even make it to the recipe step.